book/src/storage.md

```diff
@@ -0,0 +1,68 @@
+# Persistent Storage
+
+Ariel OS supports persistent storage on a number of boards.
+It provides a key–value pair store based on [sequential-storage].
+
+This storage module is responsible for setting up the the flash storage,
+and provides the functions to interact with the key–value pair store in the storage.
+
+## Use Cases
+
+Persistent storage can be used to store configuration and other variables
+across reboots and power interruptions on the microcontroller.
+
+Both primitives and compound types can be stored.
+
+## Usage
+
+### Enabling the Storage Module
+
+The storage module can be enabled by [selecting the `sw/storage` laze module][laze-modules-book],
+which enables the `storage` Cargo feature.
+
+### Using the Storage Module
+
+The [storage module] provides the necessary getters and setters
+to interact with the storage backend.
+
+Keys are always `&str` typed.
+Values are required to implement the [`serde::Serialize`][serde-serialize]
+and [`serde::Deserialize`][serde-deserialize] traits.
+Under the hood, currently the values are serialized using [postcard].
+
+Care must be taken to always read a key with the same value type
+that it was written before.
+While using a different value type for reading than for writing is never unsafe,
+it might result in bogus data.
+
+See the [example][storage-example-repo] for details on the usage.
+
+## Flash Requirements
+
+The storage module requires at least two flash pages.
+The effective storage space available is `(N - 1) * PAGE_SIZE`,
+where `N` is the number of flash pages allocated.
+
+> Currently storage is only supported on flash whose pages have a uniform size.
+
+These pages are allocated by Ariel OS after the `rodata` section in the flash
+when the module is enabled.
+
+> Updating the firmware can move and invalidate the storage pages
+  when the firmware size differs from the previous version.
```

Comment:
I was about to add "and across firmware updates" on top until I found this.

Is this something we want to fix before the release? This is a severe confusion source. (Also, it means that as long as that is in, we don't really have to bother with format stability, because chances are its' gone soon anyway, and only likely to not move if you're tweaking minor aspects).